server/service/src/plugin/mod.rs

```diff
@@ -50,9 +98,81 @@ pub trait PluginServiceTrait: Sync + Send {
             PluginInstance::bind(row);
         }
 
+        let repo = FrontendPluginRowRepository::new(&ctx.connection);
+        for row in repo.all()? {
+            self.bind_frontend_plugin(ctx, row);
+        }
+
         Ok(())
     }
 
+    fn get_frontend_plugin_file(
+        &self,
+        ctx: &ServiceContext,
+        FrontendPluginFileRequest {
+            plugin_code,
+            filename,
+        }: &FrontendPluginFileRequest,
+    ) -> Result<Vec<u8>, FrontendPluginFileRequestError> {
+        use FrontendPluginFileRequestError as Error;
+        let plugins = ctx.frontend_plugins_cache.0.read().unwrap();
+
+        let plugin = plugins
+            .get(plugin_code)
+            .ok_or(Error::CannotFindPluginCode)?;
+
+        let file_content = plugin
+            .files_content
+            .get(filename)
+            .ok_or(Error::CannotFindFile)?;
+
+        Ok(file_content.clone())
+    }
+
+    fn bind_frontend_plugin(
```

Comment:
Add plugin row to cache (also does base_64 decoding), otherwise could have done a cache on repo